---
title: "RNA-seq_Plotting"
author: "MTM"
date: "10/10/2020"
output:
  pdf_document: default
  html_document: default
---

# Loading Souce Data 

Loaded human log2FC data, normalized counts and sample guide from Posfai's Plos Pathogens paper, and lncRNA plus gene neighbor lists from Kaessman's Nature paper.
```{r}
Hs_Log2FC <- read.csv("source_data/Hs_Log2FC.csv", header = TRUE)
lncRNAs_paper <-read.csv("source_data/lncRNAs_filtered_data_Kaessmann2019.csv", header= TRUE)
norm_counts_all<-read.csv("source_data/Norm_counts_all.csv", header = TRUE)
sample_guide<-read.csv("source_data/SampleGuide_data_dds.csv", header = TRUE)

```

Check data

```{r}
str(Hs_Log2FC)
str(lncRNAs_paper)
str(norm_counts_all)
str(sample_guide)
```

# Clean up and transform dataframes

1. Transform categorical data into factors
```{r}
Hs_Log2FC$Cell.line<-as.factor(Hs_Log2FC$Cell.line)
Hs_Log2FC$time.point<-as.factor(Hs_Log2FC$time.point)
sample_guide$Cell.line<-as.factor(sample_guide$Cell.line)
sample_guide$time.groups2<-as.factor(sample_guide$time.groups2)
```

2. Rename some columns for simplicity.

```{r}
library(dplyr)
lncRNAs_paper<-rename(lncRNAs_paper, Ensembl.ID=ENSEMBL75.id)
norm_counts_all<-rename(norm_counts_all, Ensembl.ID=X )
sample_guide<-rename(sample_guide, sample.ID=X)
```

#Plotting DEGs

1. Create variables that contain threshold criteria. The lfc.cutoff is set to 0.58; remember that we are working with log2 fold changes so this translates to an actual fold change of 1.5.
```{r}
### Set thresholds
padj.cutoff <- 0.05
logfc.cutoff <- 0.58
```
2. Create vector that helps us identify the genes that meet our criteria:
```{r}
threshold<-Hs_Log2FC$adjusted.p.value<padj.cutoff & abs(Hs_Log2FC$log2FoldChange)>logfc.cutoff
```
We now have a logical vector of values that has a length which is equal to the total number of genes in the dataset. The elements that have a TRUE value correspond to genes that meet the criteria (and FALSE means it fails). How many genes are differentially expressed in the Overexpression compared to Control, given our criteria specified above? Does this reduce our results?
```{r}
length(which(threshold))
```

To add this vector to our results table we can use the $ notation to create the column on the left hand side of the assignment operator, and the assign the vector to it instead of using cbind():
```{r}
Hs_Log2FC$threshold<-threshold
```
Now we can easily subset the results table to only include those that are significant using the subset() function:
```{r}
sig_DE_all<-data.frame(subset(Hs_Log2FC,threshold==TRUE))
```

Subset lncRNAs from sig_DE_all

1. Need to clean up lncRNA table
```{r}
#check how many lncRNAs are in table from paper
length(lncRNAs_paper$Ensembl.ID)
#test for NA values
is.na(lncRNAs_paper$Ensembl.ID)
#remove na in r - remove rows - na.omit function / option
lncRNAs_paper <-na.omit(lncRNAs_paper)
#test for NA values again
is.na(lncRNAs_paper$Ensembl.ID)
```
NA values have been removed.Here I am not sure if I removed IDs that did had an NAs in some other column, but I think it doesn't matter. 

2.Subset lncRNAs

```{r}
sig_DE_lncRNA <- sig_DE_all[sig_DE_all$Ensembl.ID %in% lncRNAs_paper$Ensembl.ID,]
print(sig_DE_lncRNA)
```
#Visualizing DE lncRNAs

Install and load packages that will be needed
```{r}
install.packages("DESeq2")
install.packages("reshape")
install.packages("ggrepel")
install.packages("DEGreport")
install.packages("RColorBrewer")
install.packages("pheatmap")
# Load libraries
library(reshape)
library(ggplot2)
library(ggrepel)
library(RColorBrewer)
library(pheatmap)

```
DESeq2 and DEGreport couldn't be installed.

Before it gets more complicated, I am going to split my data per cell line.
```{r}
sig_DE_lncRNA_hepg2<-sig_DE_lncRNA["HepG2",]
sig_DE_lncRNA_huh7<-sig_DE_lncRNA["HuH7",]
```
This didn't work, so will remove them.
```{r}
rm(sig_DE_lncRNA_hepg2)
rm(sig_DE_lncRNA_huh7)
```

Try again

```{r}
sig_DE_lncRNA_hepg2<-sig_DE_lncRNA[sig_DE_lncRNA$Cell.line=="HepG2",]
sig_DE_lncRNA_huh7<-sig_DE_lncRNA[sig_DE_lncRNA$Cell.line=="HuH7",]
```
I am going to focus on HepG2 for now

Extract the normalized count values for these genes
```{r}
sig_DE_lncRNA_hepg2_counts<-norm_counts_all[sig_DE_lncRNA_hepg2,]
```
I'm going to try again picking only the column with the genes names (columns are lists)

```{r}
sig_DE_lncRNA_hepg2_counts<-norm_counts_all[sig_DE_lncRNA_hepg2$Ensembl.ID,]
remove(sig_DE_lncRNA_hepg2_counts)
```
it produced a dataframe with all NA values so I removed it.


```{r}
remove(list_sig_DE_lncRNA_hepg2)
```

```{r}
sig_DE_lncRNA_hepg2_ID<-sig_DE_lncRNA_hepg2$Ensembl.ID
```
Now, try again extracting the normalized count values for these gene

```{r}
sig_DE_lncRNA_hepg2_counts<-norm_counts_all["sig_DE_lncRNA_hepg2",]
remove(sig_DE_lncRNA_hepg2_counts)
```

Try subsetting?
```{r}
sig_DE_lncRNA_hepg2_counts<- subset(norm_counts_all, Ensembl.ID=="sig_DE_lncRNA_hepg2_ID",)
print(sig_DE_lncRNA_hepg2_counts)
```
Remove, again

```{r}
remove(sig_DE_lncRNA_hepg2_counts)
```

```{r}
sig_DE_lncRNA_hepg2_counts<-norm_counts_all["sig_DE_lncRNA_hepg2_ID",]
remove(sig_DE_lncRNA_hepg2_counts)
```
```{r}
print(sig_DE_lncRNA_hepg2_ID)
```


Try other subsetting methods
```{r}
sig_DE_lncRNA_hepg2_counts <- norm_counts_all[norm_counts_all$Ensembl.ID %in% sig_DE_lncRNA_hepg2$Ensembl.ID,]
remove(sig_DE_lncRNA_hepg2_ID)
```

It worked!

Now that we have the normalized counts for our genes for all 41 samples, to plot using ggplot(), we need to gather the counts for all samples into a single column to allow us to give ggplot the one column with the values we want it to plot.

The melt() function in the reshape R package will perform this operation and will output the normalized counts for all genes for sample 1 listed in the first 20 rows, followed by the normalized counts for sample 2 in the next 20 rows, so on and so forth.
```{r}
## use melt to modify the format of the data frame
melted_sig_DE_lncRNA_hepg2_counts <- data.frame(melt(sig_DE_lncRNA_hepg2_counts))

## check the column header in the "melted" data frame
View(melted_sig_DE_lncRNA_hepg2_counts)

## add column names that make sense
colnames(melted_sig_DE_lncRNA_hepg2_counts) <- c("gene", "samplename", "normalized_counts")
```

Now, if we want our counts colored by sample group, then we need to combine the metadata information with the melted normalized counts data into the same data frame for input to ggplot():
```{r}
## add metadata to "melted" dataframe
sample_guide$time.groups2 <- rownames(sample_guide)

melted_sig_DE_lncRNA_hepg2_counts <- merge(melted_sig_DE_lncRNA_hepg2_counts, sample_guide)
```

Remember this contains all the huh7 samples too. What a mess. I'm going to remove several columns

```{r}
melted_sig_DE_lncRNA_hepg2_counts$samplename<-NULL
melted_sig_DE_lncRNA_hepg2_counts$count<-NULL
melted_sig_DE_lncRNA_hepg2_counts$time.groups<-NULL
melted_sig_DE_lncRNA_hepg2_counts$time.groups2<-NULL
```

```{r}
str(melted_sig_DE_lncRNA_hepg2_counts$analysis.groups)
melted_sig_DE_lncRNA_hepg2_counts$analysis.groups<-as.factor(melted_sig_DE_lncRNA_hepg2_counts$analysis.groups)
str(melted_sig_DE_lncRNA_hepg2_counts$analysis.groups)
```


Subset for HepG2 data (get rid of huh7)
```{r}

melted_sig_DE_lncRNA_hepg2_counts <- melted_sig_DE_lncRNA_hepg2_counts[melted_sig_DE_lncRNA_hepg2_counts$Cell.line =="HepG2",]

```

Now I am ready for making the heatmap

```{r}
### Annotate our heatmap (optional)
annotation <- data.frame(analysis.groups=melted_sig_DE_lncRNA_hepg2_counts[,'analysis.groups'], 
                     row.names=melted_sig_DE_lncRNA_hepg2_counts[,'gene'])

### Set a color palette
heat.colors <- brewer.pal(6, "YlOrRd")

### Run pheatmap
pheatmap(melted_sig_DE_lncRNA_hepg2_counts, color = heat.colors, cluster_rows = T, show_rownames=T,
annotation= annotation, border_color=NA, fontsize = 10, scale="row",
     fontsize_row = 10, height=20)
```
```{r}

### Set a color palette
heat.colors <- brewer.pal(6, "YlOrRd")

### Run pheatmap
pheatmap(melted_sig_DE_lncRNA_hepg2_counts, color = heat.colors, cluster_rows = T, show_rownames=T,
annotation= annotation, border_color=NA, fontsize = 10, scale="row",
     fontsize_row = 10, height=20)
```
pheatmap needs a matrix, can't work with a dataframe

sig_DE_lncRNA_hepG2_counts was meant to be JUST the counts. so extract that.
```{r}
### Extract counts from sig_de_lncRNA_hepg2_counts to make it a matrix
counts_sigLncHepG2 <- sig_DE_lncRNA_hepg2_counts[,(2:41)]
```

Try heatmap again

```{r}
### Set a color palette
heat.colors <- brewer.pal(11, "PiYG")

### Run pheatmap
pheatmap(counts_sigLncHepG2, color = heat.colors, cluster_rows = T, show_rownames=F,
, border_color=NA, fontsize = 10, scale="row",
     fontsize_row = 10, height=20)
```

